src/Smalot/PdfParser/RawData/RawDataParser.php

```diff
@@ -463,6 +466,16 @@ protected function decodeXrefStream(string $pdfData, int $startxref, array $xref
                             break;
                 }
                 ++$obj_num;
+                if (isset($index_blocks)) {
+                    if (0 == --$index_blocks[0][1]) {
+                        array_shift($index_blocks);
+                        if (0 < count($index_blocks)) {
+                            $obj_num = $index_blocks[0][0];
+                        } else {
+                            unset($index_blocks);
+                        }
+                    }
+                }
```

Comment:
3 if-clauses inside each other is a code smell in my opinion.

* [ ] Please add comment which explains what this part does.
* [ ] Reduce number of if-clauses. For instance move some operations in front of the first clause and combine all their conditions?
* [ ] `if (0 == --$index_blocks[0][1]) {` is a bit problematic, because one might overlook the `--` before `$index_blocks` and assumes a different value. Such operations should be split up.

Reply:
I added comments and splitted the code with the --.
But I do not know, how I should reduce the if clauses: The first is neccessary to avoid problems, if $index_blocks does not exists. The second checks the downcounting and only if the count is 0 there must be a check with the third if whether or not there are additional entries.